### ISSUE TITLE
Advice on bulk write operations

### ISSUE THREAD

**Issue by renewooller** [RedisGraph/RedisGraph#759]
Seeking advice on how best to do mass insertions or upserts into resisgraph. 

I can't use the bulk loader because the graph key will already exist.

I realise that I can do many creations of nodes and relationships within the one query, but I'm also wondering if there's much benefit to developing support for pipelining in a client (https://redis.io/topics/pipelining) or using text file mass insertion (https://redis.io/topics/mass-insert) for these  queries, or if queries over multiple connections in parallel would suffice (EG just bluebird's promise.map with eg concurrency 100). 

Thanks!

**Comment by renewooller** [comment]
for anyone interested, ioredisgraph has support for pipelining as observed in the source code: https://github.com/Jonahss/ioredisgraph/blob/master/index.js

**Comment by renewooller** [comment]
sorry I realised I should ask this on the mailing list. FWIW, I found that for relationships, 50 +/-25 creates per query was a sweet spot and could achieve aprox 5k creates / second on my local instance (2.2gz i7). Pipelining didn't seem to have much of an affect locally (of course due to negligible latency), and degraded efficiency at > 125 requests per pipeline chunk. Having indices on the nodes was essential. So I'll start with 50 creates per query and 100 queries per pipeline.
